include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -95,24 +103,16 @@ def MuxSameCondHigh : Pat<
     (EqualTypes $x, $y), (EqualTypes $x, $z), (KnownWidth $x)
   ]>;
 
-// regreset(clock, invalidvalue, resetValue) -> reg(clock)
-def RegResetWithInvalidReset : Pat<
-  (RegResetOp $clock, (InvalidValueOp), $_, $name, $nameKind , $annotations, $inner_sym),
-  (RegOp $clock, $name, $nameKind, $annotations, $inner_sym),
-  []>;
-
-// regreset(clock, reset, invalidvalue) -> reg(clock)
-// This is handled by the `RemoveReset` pass in the original Scala code.
-def RegResetWithInvalidResetValue : Pat<
-  (RegResetOp $clock, $_, (InvalidValueOp), $name, $nameKind, $annotations, $inner_sym),
-  (RegOp $clock, $name, $nameKind, $annotations, $inner_sym),
-  []>;
-
 // regreset(clock, constant_zero, resetValue) -> reg(clock)
 def RegResetWithZeroReset : Pat<
   (RegResetOp $clock, $reset, $_, $name, $nameKind, $annotations, $inner_sym),
   (RegOp $clock, $name, $nameKind, $annotations, $inner_sym), [(ZeroConstantOp $reset)]>;
 
+// regreset(clock, constant_one, resetValue) -> reg(clock)
+def RegResetWithOneReset : Pat<
+  (RegResetOp $clock, $reset, $resetVal, $name, $nameKind, $annotations, $inner_sym),
+  (NodeOp $resetVal, $name, $nameKind, $annotations, $inner_sym), [(OneConstantOp $reset)]>;
+
```

Comment:
This is invalid when reg reset is used as sink.

Reply:
Yea, it should delete the other connects.